pkg/spoke/submarineragent/diagnose_controller.go

```diff
@@ -0,0 +1,219 @@
+				if jobLabelValue == jobName {
+					return true
+				}
+			}
+
+			return false
+		}, input.PodInformer.Informer()).
+		WithSync(c.sync).
+		ToController("SubmarinerAgentDiagnoseConfigController", input.Recorder)
+}
+
+func (c *submarinerDiagnoseConfigController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	if c.onSyncDefer != nil {
+		defer c.onSyncDefer()
+	}
+
+	addOn, err := c.addOnLister.ManagedClusterAddOns(c.clusterName).Get(constants.SubmarinerAddOnName)
+	if apiErrors.IsNotFound(err) {
+		// the addon not found, could be deleted, ignore
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	diagnoseConfig, err := c.diagnoseConfigLister.SubmarinerDiagnoseConfigs(c.clusterName).Get(constants.SubmarinerConfigName)
+	if apiErrors.IsNotFound(err) {
+
+		syncCtx.Recorder().Eventf("TestDiagnoseConfig", "Deleting Diagnose Job")
+
+		c.deleteDiagnoseJob(addOn.Spec.InstallNamespace)
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if diagnoseConfig.Status.Conditions != nil {
+		// syncCtx.Recorder().Event("TestDiagnoseConfig", "TESTING Status already set")
+		return nil
+	}
+
+	podList, err := c.podLister.Pods(addOn.Spec.InstallNamespace).List(labels.SelectorFromSet(labels.Set{jobLabel: jobName}))
+	if err != nil && !apiErrors.IsNotFound(err) {
+		syncCtx.Recorder().Event("TestDiagnoseConfig", fmt.Sprintf("failed to get diagnose podlist"))
+		return err
+	}
+	if apiErrors.IsNotFound(err) || len(podList) == 0{
+		syncCtx.Recorder().Event("TestDiagnoseConfig", "creating Diagnose Job")
+		err = c.deployDiagnoseJob(diagnoseConfig, addOn.Spec.InstallNamespace)
+		return err
+	}
+	// Pod deployed, update status
+	condition := diagnoseSuccessCondition("Pod %s available on cluster %s", podList[0].Name, addOn.ClusterName)
+	updateErr := c.updateSubmarinerDiagnoseStatus(ctx, syncCtx.Recorder(), diagnoseConfig, &condition)
+	if updateErr != nil {
+		syncCtx.Recorder().Event("TestDiagnoseConfig", "failed to update status")
+	}
+	return updateErr
+}
+
+func (c *submarinerDiagnoseConfigController) updateSubmarinerDiagnoseStatus(ctx context.Context, recorder events.Recorder,
+	config *diagnosev1alpha1.SubmarinerDiagnoseConfig, condition *metav1.Condition) error {
+	updatedStatus, updated, err := submarinerdiagnoseconfig.UpdateStatus(ctx,
+		c.diagnoseClient.SubmarineraddonV1alpha1().SubmarinerDiagnoseConfigs(config.Namespace), config.Name,
+		submarinerdiagnoseconfig.UpdateConditionFn(condition))
+
+	if updated {
+		recorder.Eventf("SubmarinerDiagnoseConfigStatusUpdated", "Updated status conditions:  %#v", updatedStatus.Conditions)
+	}
+
+	return err
+}
+
+func (c *submarinerDiagnoseConfigController) deployDiagnoseJob(config *diagnosev1alpha1.SubmarinerDiagnoseConfig,
+		namespace string) error {
+	var backOffLimit int32 = 0
+	jobSpec := &batchv1.Job{
+		ObjectMeta : metav1.ObjectMeta{
+			Name: jobName,
+			Namespace: namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
```

Comment:
Don't you also need the pod template labels, ie

```
ObjectMeta: metav1.ObjectMeta{
    Labels: map[string]string{jobLabel: jobName},
},
```

Reply:
K8s adds the label for all pods created by jobs.